components/raftstore/src/store/fsm/peer.rs

```diff
@@ -817,6 +823,36 @@ where
         self.register_raft_base_tick();
     }
 
+    fn on_unsafe_recovery_wait_apply(&mut self, commit_index: u64, counter: Arc<AtomicUsize>) {
+        if self.fsm.peer.raft_group.store().applied_index() != commit_index {
+            sleep(Duration::from_millis(1000));
```

Comment:
it will block the raftstore, please record the commit index, and check it on the update of the apply index, which is in `on_apply_res`